From 5cbcf50aa34f42ae3ec7bc83a5ca62c6454e278c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Mar 2020 13:26:37 +0200
Subject: [PATCH] [SW-10047] gaudi: pll: remove all config data

The PLLs configuration handling is moving to U-Boot in order to minimize
the preboot image.
Remove the PLLs handling from preboot.

Note: this slows down the Gaudi boot by about 8 seconds!

Change-Id: Ib593cd492cd080df384ae6daa43f8f4dce586bc4
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 22f43ff..a873e74 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -6,162 +6,12 @@
  */
 
 #include <psoc_global_conf_regs.h>
-#include <psoc_pci_pll_regs.h>
 
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
 
 #include "common.h"
-
-static struct hl_pll_cfg gaudi_pll_cfg[] = {
-	{
-		.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
-		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
-		.div_factor = { 0, 1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "IF_W", .base = mmIF_W_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "IF_E", .base = mmIF_E_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
-		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
-		.div_factor = { 0, 1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
-		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
-		.nr = 0, .nf = 35, .od = 1, .nb = 17, /* 900 */
-		.div_factor = { 0, -1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
-		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
-		.nr = 0, .nf = 35, .od = 1, .nb = 17, /* 900 */
-		.div_factor = { 0, -1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
-//		.nr = 0, .nf = 37, .od = 1, .nb = 18, /* 950 */
-		.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
-		.div_factor = { 0, 1, 4, 9 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "NIC0", .base = mmNIC0_PLL_BASE,
-		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
-		.div_factor = { 0, 4, 8, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "NIC1", .base = mmNIC1_PLL_BASE,
-		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
-		.div_factor = { 0, 4, 8, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	},
-};
-
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	int pll_num = ARRAY_SIZE(gaudi_pll_cfg);
-
-	*hl_pll_cfg = gaudi_pll_cfg;
-
-	return pll_num;
-}
 
 u32 hl_skip_pcie_init(void)
 {
